crates/revm/revm-inspectors/src/stack/maybe_owned.rs

```diff
@@ -85,86 +90,92 @@ where
             }
             MaybeOwnedInspector::Stacked(_) => {}
         }
     }
 
-    fn step_end(&mut self, interp: &mut Interpreter<'_>, data: &mut EVMData<'_, DB>) {
+    fn step_end(&mut self, interp: &mut Interpreter, context: &mut EvmContext<'_, DB>) {
         match self {
-            MaybeOwnedInspector::Owned(insp) => insp.borrow_mut().step_end(interp, data),
+            MaybeOwnedInspector::Owned(insp) => insp.borrow_mut().step_end(interp, context),
             MaybeOwnedInspector::Stacked(_) => {}
         }
     }
 
     fn call(
         &mut self,
-        data: &mut EVMData<'_, DB>,
+        context: &mut EvmContext<'_, DB>,
         inputs: &mut CallInputs,
-    ) -> (InstructionResult, Gas, Bytes) {
+    ) -> Option<(InterpreterResult, Range<usize>)> {
         match self {
-            MaybeOwnedInspector::Owned(insp) => return insp.borrow_mut().call(data, inputs),
+            MaybeOwnedInspector::Owned(insp) => return insp.borrow_mut().call(context, inputs),
             MaybeOwnedInspector::Stacked(_) => {}
         }
 
-        (InstructionResult::Continue, Gas::new(0), Bytes::new())
+        Some((
+            InterpreterResult {
+                result: InstructionResult::Continue,
+                gas: Gas::new(0),
+                output: Bytes::new(),
+            },
+            0..0,
+        ))
     }
 
     fn call_end(
         &mut self,
-        data: &mut EVMData<'_, DB>,
-        inputs: &CallInputs,
-        remaining_gas: Gas,
-        ret: InstructionResult,
-        out: Bytes,
-    ) -> (InstructionResult, Gas, Bytes) {
+        context: &mut EvmContext<'_, DB>,
+        result: InterpreterResult,
+    ) -> InterpreterResult {
         match self {
-            MaybeOwnedInspector::Owned(insp) => {
-                return insp.borrow_mut().call_end(data, inputs, remaining_gas, ret, out)
-            }
+            MaybeOwnedInspector::Owned(insp) => return insp.borrow_mut().call_end(context, result),
             MaybeOwnedInspector::Stacked(_) => {}
         }
-        (ret, remaining_gas, out)
+        result
     }
 
     fn create(
         &mut self,
-        data: &mut EVMData<'_, DB>,
+        context: &mut EvmContext<'_, DB>,
         inputs: &mut CreateInputs,
-    ) -> (InstructionResult, Option<Address>, Gas, Bytes) {
+    ) -> Option<(InterpreterResult, Option<Address>)> {
         match self {
-            MaybeOwnedInspector::Owned(insp) => return insp.borrow_mut().create(data, inputs),
+            MaybeOwnedInspector::Owned(insp) => return insp.borrow_mut().create(context, inputs),
             MaybeOwnedInspector::Stacked(_) => {}
         }
 
-        (InstructionResult::Continue, None, Gas::new(0), Bytes::default())
+        Some((
+            InterpreterResult {
+                result: InstructionResult::Continue,
+                gas: Gas::new(0),
+                output: Bytes::new(),
+            },
+            None,
+        ))
```

Comment:
Same here, it should be None as in here: https://github.com/paradigmxyz/reth/pull/5489/files/880fd26ca13093810afc7f6d2e1e9617b44c0ecb#r1398242606